urllib3/request.py

```diff
@@ -1,13 +1,51 @@
 from __future__ import absolute_import
-try:
-    from urllib.parse import urlencode
-except ImportError:
-    from urllib import urlencode
+
+import logging
 
 from .filepost import encode_multipart_formdata
+from .exceptions import MaxRetryError
+
+from .packages.six.moves.urllib.request import Request as _Request
+from .packages.six.moves.urllib.parse import urlencode
+from .packages.six.moves.urllib.parse import urljoin
+
+__all__ = ['RequestMethods', 'Request']
+
+log = logging.getLogger(__name__)
 
 
-__all__ = ['RequestMethods']
+class Request(_Request):
+    """
+    Currently used as a shim to allow us to work with the stdlib cookie
+    handling, which expects a `urllib.request.Request`-like object.
+    """
+    def __init__(self, *args, **kwargs):
+        kwargs.pop('method', None)
```

Comment:
Given that we're doing this only for its side effects, let's swap to `del kwargs['method']`